### ISSUE TITLE
ListView Drag&Drop Example

### ISSUE THREAD

**Issue by Luk3s** [flutter/flutter#13651]
I struggle to find any example of a Drag&Drop ListView, with the list items being a Row Widget made of at least two other Widgets and one of them inside an Expanded (I can see people having problems with the feedback Widget to keep the same dimension as the child of Draggable). Also having the other list items moving up or down while one is dragged over would be very helpful. And last, it would be useful to have the list scrolling up and down depending on where the draggable item is dragged.

Something like this https://www.youtube.com/watch?v=tNgevYpyA9E (for the list view only, the video contains some examples for GridView as well)

**Comment by Alexays** [comment]
Look at this https://github.com/rxlabz/orderable_stack, it's a good start. ;)

**Comment by Luk3s** [comment]
Thanks, this is indeed a good start, it hasn't been updated in a while but there are some interesting pieces of code that can be used to achieve what I was planning to do, though it would be helpful to have an official example because I feel that this is a very popular interaction with lists.

**Comment by elgansayer** [comment]
Did you finish this or get anywhere and be willing to share?

**Comment by marchampson** [comment]
I'm working on this at the moment, @Luk3s if you did get anywhere would be great if you're willing to share as @elgansayer says. If I get anywhere I'll post a link. Thanks

**Comment by Norbert515** [comment]
@marchampson how's progress going? I also need to implement this & before starting from scratch, I'd like to ask you if I can help.

**Comment by marchampson** [comment]
Hi @Norbert515 I got it sorted, not near my laptop till tomorrow. If you can wait I’ll upload a version to github.

**Comment by marchampson** [comment]
Here you go @Norbert515, hope this helps, just clone or fire up a new Flutter project yourself and copy the contents of lib/main.dart into the relevant file. Should hopefully get you going. I should mention the good folks on Gitter: https://gitter.im/flutter/flutter?source=orgpage helped along the way.

https://github.com/marchampson/FluterSortableListView

**Comment by Norbert515** [comment]
Thanks, @marchampson I will additionally try to implement moving up down, scrolling and animations.

**Comment by marchampson** [comment]
@Norbert515 When you say moving up and down do you mean as you hover a row over another? Similar to normal iOS/Android behaviour? I did look at that but because this had to be implemented using DragTargets then once they move, they're no longer an active target. I guess what you could do is when `onWillAccept` is activated, you could move the activated target away by 50% and highlight it with an animation? Then drop it back with `onLeave`. If it doesn't move completely (like iOS) then it will still be a target. This certainly isn't behaving exactly as I wanted but it did solve the sortable issue.
If you get anywhere with the animations etc please feel to open a pull request on the repo. 
Thanks

**Comment by Norbert515** [comment]
@marchampson I rewrote pretty much everything and made a new repo: https://github.com/Norbert515/flutter_list_drag_and_drop
Most things already work, it is still work in progress tho.
My current implementation uses the `onWillAccept` to create a placeholder item(and the `onLeave ` to destroy it) with the correct size and `onAccept` simply replaces the placeholder with the actual item.

**Comment by marchampson** [comment]
@Norbert515 A good start, I think you might have a few too many `rows.removeAt(index)` floating around. After 3 drags I'm down to 3 rows remaining but I see where you're going with it. Keep me posted.

**Comment by Norbert515** [comment]
@marchampson that's weird, I'll look into it.
I currently have this strange glitch when a little item gets dragged passed a big item (small over small/ big over big/small works perfectly). Would you mind taking a quick look at the code and tell me your thoughts on that, I've been staring at the code for too long haha.

**Comment by Norbert515** [comment]
@marchampson yes that's because there is not `DropTarget`, will probably have to place an invisible widget on the top/bottom of the list to catch these cases.

**Comment by myleftshoe** [comment]
Thanks for this repo @Norbert515!!! It's looking good, exactly what I need. Can't wait for the update.  Would love to contribute but I don't know flutter well enough yet.

**Comment by knopp** [comment]
My take on [iOS-like reorderable listview](https://github.com/knopp/flutter_reorderable_list) in case someone finds it useful (with animations).

**Comment by AppleEducate** [comment]
@knopp You are a lifesaver. I have been looking for that functionality since I started with flutter. thanks

**Comment by zoechi** [comment]
See also #18374

**Comment by lukepighetti** [comment]
@knopp Any chance you'll put this on pub?

**Comment by knopp** [comment]
@lukepighetti https://pub.dartlang.org/packages/flutter_reorderable_list

**Comment by parth22** [comment]
@Norbert515 can i perform animation in this list view like when i delete items from this drag and drop list view if yes, then it will very help full.

**Comment by mostafa8026** [comment]
@Allan-Nava Do you find any solution of drag-drop between two lists?

**Comment by VladyslavBondarenko** [comment]
@marchampson @Allan-Nava 
I just noticed proposal about it, you may be interested to thumb it up https://github.com/flutter/flutter/issues/53751

**Comment by mehtahardikr** [comment]
I want to develop like  drag one listitem to another list which may be empty. if dragitem placed on another list's  particular item then dragged item will be added to that index  of list to which it dragged and same thing will be applicable on same list also. i tried but i am  not able to make it like scroll listview while dragitem selected and  there are only 3 visible items. 

I want to develop like this:  https://i.stack.imgur.com/bdn1I.gif   

Any help appreciated.

**Comment by Piinks** [comment]
(Triage) It looks like we can close this now that we have ReorderableList and SliverReorderableList. If there is still something unaddressed here, please file a new issue. Thank you!
